argmin/src/solver/gradientdescent/steepestdescent.rs

```diff
@@ -65,13 +65,16 @@ where
         problem: &mut Problem<O>,
         mut state: IterState<P, G, (), (), F>,
```

Comment:
This does not need to be mutable anymore. Then the clippy checks in the CI should pass.

Reply:
Done, I just force pushed since it was a small change